server/consumer.go

```diff
@@ -1238,7 +1247,8 @@ func (o *Consumer) processNextMsgReq(_ *subscription, c *client, _, reply string
 	if o.replay {
 		o.waiting.add(&wr)
 		o.mu.Unlock()
-		mset.signalConsumers()
+		o.signalNewMessages()
```

Comment:
In some places this is invoked inside of consumer lock, but here it is outside. Note that in signalNewMessages() you are accessing consumer's field that may or may not be immutable, not sure.

Reply:
Yes understand. I rely on Go's race detector for things setup once at creation time and never changed such that locks are not required if the underlying pointer is safe like channels.

Reply:
So unless consumer object is created in some tests without proper call to AddConsumer - and store is possibly not set, maybe we should remove the check to `o.store == nil` in signalNewMessages() to simply avoid confusion or second guessing.